projects/plugins/boost/app/modules/cache/Page_Cache.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+use Automattic\Jetpack_Boost\Contracts\Is_Always_On;
+use Automattic\Jetpack_Boost\Contracts\Pluggable;
+
+class Page_Cache implements Pluggable, Is_Always_On {
+	/*
+	 * Sets up the advanced-cache.php file and if that works, adds the WP_CACHE
+	 * define to wp-config.php
+	 * These are used by WordPress to load the caching system before most of
+	 * WordPress is loaded.
+	 */
+	public function setup() {
+		if ( is_admin() && isset( $_GET['page'] ) && $_GET['page'] === 'jetpack-boost' ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			require_once ABSPATH . '/wp-admin/includes/file.php';
+			WP_Filesystem();
+			$result = $this->create_advanced_cache();
+			if ( $result === true ) {
+				$result = $this->add_wp_cache_define();
+				if ( $result !== true ) {
+					return $result;
+				}
+			} else {
+				return $result;
+			}
+
+			return true;
```

Comment:
From what I could find, we don't handle returns for a module setup.

Since this is the first module that needs a setup (I think), maybe we need to expand it in a way? Like check if folders are writable, create files, add defines, etc.

Though I don't think we need to do it now.

Reply:
yep. Definitely needs to be done.

Reply:
Can we turn this into a ticket on the board / TODO list for Boost cache?

Reply:
Done. Added in [#35215](https://github.com/Automattic/jetpack/issues/35215).